### PR TITLE
Add ability to define TestGrid TestGroup days_of_results through ProwJob annotations.

### PR DESCRIPTION
**Pull request by Ressetkk**
As I mentioned in my post in sig-testing Slack channel I've added the functionality of defining `days_of_result` field to the `configurator` and added new annotation to the tests.

Link to the conversation: https://kubernetes.slack.com/archives/C09QZ4DQB/p1594986830198500

### CODE REVIEW

File: testgrid/cmd/configurator/prow.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ const testgridEmailAnnotation = "testgrid-alert-email"
 const testgridNumColumnsRecentAnnotation = "testgrid-num-columns-recent"
 const testgridAlertStaleResultsHoursAnnotation = "testgrid-alert-stale-results-hours"
 const testgridNumFailuresToAlertAnnotation = "testgrid-num-failures-to-alert"
+const testgridDaysOfResultAnnotation = "testgrid-days-of-result"
```
Comment (michelle192837) [inline testgrid/cmd/configurator/prow.go:43]: nit: Should be 'results' and not 'result', here and elsewhere

Comment (Ressetkk) [reply testgrid/cmd/configurator/prow.go]: Renamed it.

**Pull request merged**